barcalendar.py

```diff
@@ -245,6 +249,51 @@ def column_marker(self, column):
     def write(self):
         self.epilog()
 
+def get_defaults_from_tutor():
+    """
+    Fetches default configurations from tutor repository.
+    Returns:
+        object: Default configurations as Python object
+    """
+    url = "https://raw.githubusercontent.com/overhangio/tutor/master/tutor/templates/config/defaults.yml"
+    while True:
+        try:
+            resp = requests.get(url, timeout=10)
+        except requests.RequestException as exc:
+            print(f"Couldn't fetch {url}: {exc}")
+            return None
+        if resp.status_code == 429:
+            wait = int(resp.headers.get("Retry-After", 10))
+            time.sleep(wait + 1)
+        else:
+            break
+
+    if resp.status_code == 200:
+        return safe_load(resp.text)
+    return None
```

Comment:
For many of the info functions, it would be better to raise an exception than to return None.  If a problem occurs, and they return None, we'll have a mysterious error elsewhere in the program.

Reply:
Added a `raise RuntimeError` if `resp.status_code` is anything but `200`. 

Also in the `except` block above, I re-raised the `RequestException` error. Let me know if this makes sense.

Reply:
Requests has a helper method: https://requests.readthedocs.io/en/latest/api/#requests.Response.raise_for_status .  It will raise an exception based on the status code.

Reply:
Thanks, did not know about this. I've incorporated it.